M10/yul.sol

```diff
@@ -0,0 +1,54 @@
+pragma solidity ^0.8.4;
+
+contract BitWise {
+    // count the number of bit set in data.  i.e. data = 7, result = 3
+    function countBitSet(uint8 data) public pure returns (uint8 result) {
+        for (uint i = 0; i < 8; i += 1) {
+            if (((data >> i) & 1) == 1) {
+                result += 1;
+            }
+        }
+    }
+
+    function countBitSetAsm(uint8 data) public pure returns (uint8 result) {
+        // replace following line with inline assembly code
+        assembly {
+            for {
+                let i := 0
+            } lt(i, 8) {
+                i := add(i, 1)
+            } {
+                let s := shr(i, data)
+                let r := and(s, 1)
+                if eq(r, 1) {
+                    result := add(result, 1)
```

Comment:
* [ ] assign the number of bits set to a variable declared within assembly block and put it to memory to return the result.